packages/cli/test/scripts/update.test.js

```diff
@@ -71,142 +74,188 @@ describe('update script', function() {
+      contractName: 'ImplV1',
       network,
       txParams,
       networkFile: this.networkFile,
     });
     this.proxy3 = await createProxy({
-      contractAlias: 'WithLibraryImpl',
+      contractName: 'WithLibraryImplV1',
       network,
       txParams,
       networkFile: this.networkFile,
     });
   };
 
-  const bumpVersion = async function() {
+  const bumpVersionAndPush = async function() {
+    const contracts = ['ImplV1', 'WithLibraryImplV1', 'ImplV2', 'WithLibraryImplV2'];
     await bump({ version: version2, txParams, projectFile: this.projectFile });
-    const newContractsData = [
-      { name: 'ImplV2', alias: 'Impl' },
-      { name: 'WithLibraryImplV2', alias: 'WithLibraryImpl' },
-    ];
     await add({
-      contractsData: newContractsData,
+      contracts,
       projectFile: this.projectFile,
     });
-    await push({ network, txParams, networkFile: this.networkFile });
 
-    this.implV2Address = this.networkFile.contract('Impl').address;
-    this.withLibraryImplV2Address = this.networkFile.contract('WithLibraryImpl').address;
+    await push({ contracts, network, txParams, networkFile: this.networkFile });
+
+    this.implV2Address = this.networkFile.contract('ImplV2').address;
+    this.withLibraryImplV2Address = this.networkFile.contract('WithLibraryImplV2').address;
+  };
+
+  const stubUpdate = (mappingToNames, mappingToAddresses) => {
+    sinon.stub(ContractManager.prototype, 'getContractClass').callsFake(function(packageName, contractName) {
+      return ContractManager.prototype.getContractClass.wrappedMethod.apply(this, [
+        packageName,
+        mappingToNames[contractName] ?? contractName,
+      ]);
+    });
+    sinon
+      .stub(ProxyAdminProject.prototype, 'getImplementation')
+      .callsFake(function({ packageName, contractName, contract }) {
+        return ProxyAdminProject.prototype.getImplementation.wrappedMethod.apply(this, [
+          {
+            packageName,
+            contractName: mappingToNames[contractName] ?? contractName,
+            contract,
+          },
+        ]);
+      });
+    sinon.stub(AppProject.prototype, 'getImplementation').callsFake(function({ packageName, contractName, contract }) {
+      return AppProject.prototype.getImplementation.wrappedMethod.apply(this, [
+        {
+          packageName,
+          contractName: mappingToNames[contractName] ?? contractName,
+          contract,
+        },
+      ]);
+    });
+    sinon
+      .stub(ProxyAdminProject.prototype, '_getOrDeployOwnImplementation')
+      .callsFake(function(contract, contractName, redeployIfChanged) {
+        return mappingToAddresses[contractName]
+          ? Promise.resolve(mappingToAddresses[contractName])
+          : ProxyAdminProject.prototype._getOrDeployOwnImplementation.wrappedMethod.apply(this, [
+              contract,
+              contractName,
+              redeployIfChanged,
+            ]);
+      });
+  };
+
+  const restoreUpdate = () => {
```

Comment:
It is actually [depreciated](https://github.com/sinonjs/sinon/issues/1390) but we use old version of Sinon. If we upgrade Sinon our tests would fail 😅